25_exceptions_lab/03.py

```diff
@@ -1,4 +1,28 @@
-"""
-Write code to make the following unit test pass
-"""
+#!/usr/bin/python
 
+import unittest
+
+class InvalidImageExt(Exception): pass
+
+class TestImageFile(unittest.TestCase):
+    def test_good_ext(self):
+        try:
+            img = ImageFile("file.png")
+        except InvalidImageExt:
+            self.fail("png should be a valid file extension")
+
+    def test_bad_ext(self):
+        with self.assertRaises(InvalidImageExt):
+            img = ImageFile("file.mp3")
+
+class ImageFile(object):
+	def __init__(self, name):
+		self.name = name
+		
```

Comment:
Great. I'd use something more generic such as:

```
class ImageFile(object):
    valid_extensions = ['png', 'jpg', 'jpeg']

    def __init__(self, name):
        if any(name.endswith('.' + ext) for ext in ImageFile.valid_extensions):
            # valid image file
            print 'valid'
        else:
            print 'invalid'
```

Notice the use of `any` to check if any value from the list comprehension is True